### PR TITLE
Feat/icon tabs

### PR DESCRIPTION
**Pull request by ecni19**
# Tabs with icons

- [Figma](https://www.figma.com/file/RGYLmkueoutDtl6vQNZTJU/Tabs-with-icons)
- `td-icon-tab` inherits from TabBase and `td-icon-tab-bar` inherits from TabBarBase.
- Secondary text is optional but icon and label are required (all passed via properties).
- Text and icon darken when selected.

IconTabDemo

Overflow is accounted for. 

Screen Shot 2022-08-18 at 9 56 53 AM

### CODE REVIEW

**Review by adamnel — APPROVED** [review]

**Comment by adamnel** [comment]
@mikespudwebb @owilliams320 Want to review or shall I merge? Looks good on my end

File: lib/components/src/icon-tabs/icon-tab.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+declare global {
+  interface HTMLElementTagNameMap {
+   'td-icon-tab': CovalentIconTabBase;
+  }
+}
+
+@customElement('td-icon-tab')
+export class CovalentIconTabBase extends TabBase {
+  static override styles = [styles, tabBarStyles];
+  @property({type: String}) secondLine = 'Label';
```
Comment (mikespudwebb) [inline lib/components/src/icon-tabs/icon-tab.ts:15]: should this variable be blank by default?

**Review by mikespudwebb — COMMENTED** [review]
One small question, otherwise, looks good.

File: lib/components/src/icon-tabs/icon-tab.ts (hunk 1 above)
Comment (ecni19) [reply lib/components/src/icon-tabs/icon-tab.ts:15]: Yeah it should be

**Review by mikespudwebb — APPROVED** [review]
LGTM

**Pull request merged**